services/device-registry-base/src/test/java/org/eclipse/hono/service/management/device/AbstractDeviceManagementSearchDevicesTest.java

```diff
@@ -0,0 +1,244 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.management.device;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.eclipse.hono.util.Constants;
+import org.junit.jupiter.api.Test;
+
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Future;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * As suite of tests for verifying implementations of the Device management's 
+ * search devices operation.
+ * <p>
+ * Concrete subclasses need to provide the service implementations under test
+ * by means of implementing the {@link #getDeviceManagementService()} method.
+ */
+public abstract class AbstractDeviceManagementSearchDevicesTest {
+
+    /**
+     * The tenant used in tests.
+     */
+    protected static final String TENANT = Constants.DEFAULT_TENANT;
+
+    /**
+     * Gets device management service being tested.
+     *
+     * @return The device management service
+     */
+    public abstract DeviceManagementService getDeviceManagementService();
+
+    private String randomDeviceId() {
+        return UUID.randomUUID().toString();
+    }
+
+    /**
+     * Verifies that a request to search devices fails with a {@value HttpURLConnection#HTTP_NOT_FOUND}
+     * when no matching devices are found.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void testSearchDevicesWhenNoDevicesAreFound(final VertxTestContext ctx) {
+        final int pageSize = 10;
+        final int pageOffset = 0;
+        final Filter filter = new Filter("/enabled", false);
```

Comment:
how about using a random value for an *ext* property in order to make sure that the result set will always be scoped to devices that have this property? This should also safeguard against any side effects that might occur when running test cases concurrently.